File: desafio-api/src/main/java/com/capgemini/service/ContaService.java
package com.capgemini.service;

import java.util.List;

import com.capgemini.model.Conta;
import com.capgemini.model.ResponseModel;
import com.capgemini.service.dto.ContaDTO;

public interface ContaService {
	
	Conta getSaldo(Long contaId);

	ResponseModel sacar(ContaDTO conta);

	List<Conta> getContas();

	ResponseModel depositar(ContaDTO conta);
}

File: desafio-angular/src/app/conta/conta.service.ts
import {Injectable} from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';

import { Conta } from '../model/conta.model';
import { Response } from '../model/reponse.model';

const httpOptions = {
  headers: new HttpHeaders({ 'Content-Type': 'application/json' })
};

@Injectable()
export class ContaService {

  constructor(private http:HttpClient) {}

  	private contaUrl = 'http://localhost:8080/api';

  public getSaldo(conta) {
    return this.http.get<Conta>(this.contaUrl + "/obterSaldo/"+ conta.id);
  }

  public getContas() {
    return this.http.get<Conta[]>(this.contaUrl + "/contas");
  }

  public sacar(conta) {
    return this.http.put<Response>(this.contaUrl + "/sacar", conta);
  }

  public depositar(conta) {
    return this.http.put<Response>(this.contaUrl + "/depositar", conta);
  }
  
}